File: Sources/AdbShell/Models/PortForwardRule.swift
import Foundation

/// Одно правило проброса порта adb — либо forward (соединения на Mac
/// пробрасываются на устройство), либо reverse (соединения на устройстве
/// пробрасываются на Mac). hostSpec/deviceSpec — в формате adb, например
/// "tcp:8080".
struct PortForwardRule: Identifiable, Hashable {
    enum Direction: String {
        case forward
        case reverse
    }

    var id: String { "\(direction.rawValue)-\(hostSpec)-\(deviceSpec)" }
    let direction: Direction
    /// Порт/сокет на Mac.
    let hostSpec: String
    /// Порт/сокет на устройстве.
    let deviceSpec: String
}

File: desktop/src/main/adb/types/PortForwardRule.ts
// Порт Sources/AdbShell/Models/PortForwardRule.swift

export type ForwardDirection = 'forward' | 'reverse';

export interface PortForwardRule {
  direction: ForwardDirection;
  /** Порт/сокет на Mac/Windows (host). */
  hostSpec: string;
  /** Порт/сокет на устройстве. */
  deviceSpec: string;
}
